src/amqp/amqpnotificationinterface.cpp

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2017 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "amqpnotificationinterface.h"
+#include "amqppublishnotifier.h"
+
+#include "version.h"
+#include "main.h"
+#include "streams.h"
+#include "util.h"
+
+// AMQP 1.0 Support
+//
+// The boost::signals2 signals and slot system is thread safe, so CValidationInterface listeners
+// can be invoked from any thread.
+//
+// Currently signals are fired from main.cpp so the callbacks should be invoked on the same thread.
+// It should be safe to share objects responsible for sending, as they should not be run concurrently
+// across different threads.
+//
+// Developers should be mindful of where notifications are fired to avoid potential race conditions.
+// For example, different signals targeting the same address could be fired from different threads
+// in different parts of the system around the same time.
+//
+// Like the ZMQ notifiation interface, if a notifier fails to send a message, the notifier is shut down.
+//
+
+AMQPNotificationInterface::AMQPNotificationInterface()
+{
+}
+
+AMQPNotificationInterface::~AMQPNotificationInterface()
+{
+    Shutdown();
+
+    for (std::list<AMQPAbstractNotifier*>::iterator i=notifiers.begin(); i!=notifiers.end(); ++i)
+    {
```

Comment:
Oh I see: the ZMQ code was not consistent with the overall Bitcoin code style in this respect, and this PR is adapting the ZMQ code. I'm fine with cleaning it up later if you prefer.